src/Integration/LocalServices/ErrorListInfoBarController.cs

```diff
@@ -169,26 +162,36 @@ private void CancelQualityProfileProcessing()
 
         internal /*for testing purposes*/ void ProcessSolutionBinding()
         {
-            // No need to do anything if by the time got here the solution was closed (or unbound)
-            var mode = this.configProvider.GetConfiguration().Mode;
-            if (!mode.IsInAConnectedMode())
+            // We could be on a UI thread so any unhandled exceptions will crash VS.
```

Comment:
I haven't checked that the other entry points in this are "safe" yet.
I'll create a separate PR if they are not.